master/static/srv/agent_setup_script.sh.template

```diff
@@ -40,6 +43,7 @@ docker run --init --name determined-agent {{.LogOptions}} \
     -e DET_MASTER_HOST="{{.MasterHost}}" \
     -e DET_MASTER_PORT="{{.MasterPort}}" \
     -e DET_RESOURCE_POOL="{{.ResourcePool}}" \
+    -e DET_SLOT_TYPE=gpu \
```

Comment:
blocking: shouldn't this be set based on `{{.AgentUseGPUs}}`?

Reply:
No. It should always use `gpu`. If there are no GPUs, we should create no slots since we treat slots as exclusive to tasks and treat zero-slot tasks as special cases. If it uses `cpu`, it would create a slot for all the CPUs on the agent.

Reply:
Do we plan on setting this to `cpu` for CPU-only pools in the future?

Reply:
No. we will always set it to be `cpu` only when you want to train on CPUs and want the CPUs to be exclusive to the tasks that run on those CPUs.

Reply:
In your PR summary it says:

> in order to support a CPU resource pool when we support multiple resource pools, we need to make slot type configurable for agents.

but if you are saying we are not going to set it for CPU-only pools I am not sure I understand why we are adding this?

Reply:
Ahh. I am not precise on the terms in the description. A CPU resource pool actually means a zero-slot resource pool since we don't consider training on CPUs is explicitly supported.

Reply:
Ok I am still a bit unclear on why we needed to add this if we don't plan on support CPU-training in a first class way, since the prior implementation would do exactly what we get now by setting it to GPU always.

Reply:
By "prior implementation", do you mean the implementation this PR is building on top of? If you mean that, the prior implementation always use the `auto` mode. If there are no GPUs, it always creates a slot for each agent. For zero-slot resource pools, we don't want it to show any slots.

Reply:
Ok this makes perfect sense now 😄 . Can you add exactly this to the PR summary.

Reply:
done clarifying the terms in the description. I thought for users zero-slot is confusing. Do you get any better ideas?